### ISSUE TITLE
Test-dependencies side-affect linting in server development environments

### ISSUE THREAD

**Issue by kevin-bates** [elyra-ai/elyra#2670]
A few of us have noticed linting failures within our development that are not seen during CI.  The failures are occurring in PR #2656 on the following statements stating the imports are not in alphabetical order: 
https://github.com/elyra-ai/elyra/blob/0077dddefd126700a88528c2c0bd4e6ebb993a94/elyra/pipeline/pipeline_definition.py#L23-L24
If the statements are reversed, a different failure occurs.

These issues are not seen during CI.  

On closer examination, the issue does not occur in fresh conda environments UNTIL `make test-dependencies` has run.  This is why CI is passing, and dev environments (eventually) become tainted.

Also, note that we have removed the capture of version information in use by CI in PR https://github.com/elyra-ai/elyra/pull/1671!  This, coupled with the fact that we use `-q` on app `pip install` commands means **we have no idea which versions of packages are in use** during CI!  As a result, we should minimally capture version information prior to running tests in CI.

**To Reproduce**
1. Create a new conda env
2. Checkout the branch corresponding to PR #2656 so the import issue is present
3. Build the server using `make clean install` or your favorite targets.  This will invoke linting and it will pass
4. Run `make test-dependencies`, then `make lint-server` and the failure will occur.

**Screenshots or log output**
I captured python versions using `pip freeze` before and after `make test-dependencies` and the results follow.
<details><summary>Dependencies PRE and POST test-dependencies</summary>
<pre>
$ diff pre-test-deps post-test-deps 
22a23,24
> contextlib2==21.6.0
> d2to1==0.2.12.post1
28a31
> distlib==0.3.4
30a34
> elyra-examples-kfp-catalog==0.1.0
31a36
> execnet==1.9.0
33a39
> filelock==3.6.0
36a43
> git-python==1.0.3
47a55
> hacking==0.5.4
49a58,59
> importlib-resources==5.7.1
> iniconfig==1.1.1
80a91
> mock==4.0.3
95a107,108
> path==16.4.0
> path.py==12.5.0
96a110
> pbr==5.8.1
106a121
> py==1.11.0
120a136,141
> pytest==7.1.1
> pytest-console-scripts==1.3.1
> pytest-fixture-config==1.7.0
> pytest-shutil==1.7.0
> pytest-tornasync==0.6.0.post2
> pytest-virtualenv==1.7.0
127a149
> requests-mock==1.9.3
129a152
> requests-unixsocket==0.3.0
157a181
> virtualenv==20.14.1
164a189
> zipp==3.8.0
</pre>
</details>

Given other issues we've had surrounding the `hacking` package, I'm inclined to start there for what is causing the lining behaviors to change.

**Expected behavior**
We should have consistent results throughout the build process.

**Comment by kevin-bates** [comment]
:smile: if I `pip uninstall hacking` to deinstall `hacking == 0.5.4` `make lint-server` works!  
Then, running `pip install hacking`, I get the following results...
```
Installing collected packages: pyflakes, pycodestyle, flake8, hacking
  Attempting uninstall: pyflakes
    Found existing installation: pyflakes 2.4.0
    Uninstalling pyflakes-2.4.0:
      Successfully uninstalled pyflakes-2.4.0
  Attempting uninstall: pycodestyle
    Found existing installation: pycodestyle 2.8.0
    Uninstalling pycodestyle-2.8.0:
      Successfully uninstalled pycodestyle-2.8.0
  Attempting uninstall: flake8
    Found existing installation: flake8 4.0.1
    Uninstalling flake8-4.0.1:
      Successfully uninstalled flake8-4.0.1
ERROR: pip's dependency resolver does not currently take into account all the packages that are installed. This behaviour is the source of the following dependency conflicts.
autopep8 1.6.0 requires pycodestyle>=2.8.0, but you have pycodestyle 2.6.0 which is incompatible.
Successfully installed flake8-3.8.4 hacking-4.1.0 pycodestyle-2.6.0 pyflakes-2.2.0
```
Note the huge version discrepancy on `hacking` (`0.5.4` to `4.1.0`) along with a downgrade of `flake8` (`4.0.1` to `3.8.4`), `pycodestyle` (`2.8.0` to `2.6.0`) nd `pyflakes` (`2.4.0` to `2.2.0`).

`make lint-server` then produces:
```
$ make lint-server
flake8 elyra
elyra/pipeline/pipeline_definition.py:24:1: H306: imports not in alphabetical order (elyra.pipeline.pipeline_constants, elyra.pipeline.pipeline.operation)
make: *** [lint-server] Error 1
```

**Comment by kevin-bates** [comment]
The dependency on `hacking` was added in this commit: https://github.com/elyra-ai/elyra/commit/c3df0c62a1c33560125058c96c708fee4abaf737 and [is a set of flake8 plugins](https://docs.openstack.org/hacking/latest/user/usage.html) that test and enforce the [OpenStack StyleGuide](https://docs.openstack.org/hacking/latest/user/hacking.html#styleguide)

@ajbozarth - was the dependency on `hacking` added to lint copyright headers?  Any idea why the initial install using `make test-dependencies` lays down version `0.5.4`, but its manual re-install yields `4.1.0` along with downgrades to other lint-related dependencies?

Given this imposes downgrades of several packages, even if we set a floor of `hacking >= 4.1.0`, I think we should consider its removal.
